File: gpiApp/gpiApp/src/main/java/com/example/gpiApp/service/TeamService.java
package com.example.gpiApp.service;

import com.example.gpiApp.dto.TeamDTO;
import com.example.gpiApp.dto.UserDTO;

import java.util.List;
import java.util.Optional;

public interface TeamService {
    TeamDTO createTeam(TeamDTO teamDTO);
    
    TeamDTO updateTeam(Long teamId, TeamDTO teamDTO);
    
    void deleteTeam(Long teamId);
    
    Optional<TeamDTO> getTeamById(Long teamId);
    
    List<TeamDTO> getAllTeams();
    
    List<TeamDTO> getActiveTeams();
    
    List<TeamDTO> getTeamsByLeader(Long leaderId);
    
    List<TeamDTO> getTeamsByMember(Long memberId);
    
    TeamDTO addMemberToTeam(Long teamId, Long userId);
    
    TeamDTO removeMemberFromTeam(Long teamId, Long userId);
    
    List<UserDTO> getTeamMembers(Long teamId);
    
    long countActiveTeams();
    
    boolean existsByTeamName(String teamName);
    
    TeamDTO assignTeamLeader(Long teamId, Long leaderId);
} 

File: frontend/src/app/core/services/team.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';

export interface Team {
  id?: number;
  name: string;
  projectId: number;
  description?: string;
  members?: any[];
}

@Injectable({
  providedIn: 'root'
})
export class TeamService {
  private basePath = '/teams';

  constructor(private apiService: ApiService) {}

  getAllTeams(): Observable<Team[]> {
    return this.apiService.get<Team[]>(this.basePath);
  }

  getTeamById(id: number): Observable<Team> {
    return this.apiService.get<Team>(`${this.basePath}/${id}`);
  }

  getTeamsByProject(projectId: number): Observable<Team[]> {
    return this.apiService.get<Team[]>(`${this.basePath}/project/${projectId}`);
  }

  getTeamsByMember(userId: number): Observable<Team[]> {
    return this.apiService.get<Team[]>(`${this.basePath}/member/${userId}`);
  }

  createTeam(team: Team): Observable<Team> {
    return this.apiService.post<Team>(this.basePath, team);
  }

  updateTeam(id: number, team: Team): Observable<Team> {
    return this.apiService.put<Team>(`${this.basePath}/${id}`, team);
  }

  addMemberToTeam(teamId: number, userId: number): Observable<void> {
    return this.apiService.post<void>(`${this.basePath}/${teamId}/members/${userId}`, {});
  }

  removeMemberFromTeam(teamId: number, userId: number): Observable<void> {
    return this.apiService.delete<void>(`${this.basePath}/${teamId}/members/${userId}`);
  }

  deleteTeam(id: number): Observable<void> {
    return this.apiService.delete<void>(`${this.basePath}/${id}`);
  }
}
